File: mysql/sample_mysql.pl
#!/usr/bin/perl
use strict;
use warnings;
 
#-- Perl script that uses the "mysql" client utility to
#-- execute an SQL command

#-- Name of the database involved
my $database = $ARGV[0];
if (not defined $database)
{
  print "**Error: missing database name argument\n";
  print "  I quit!\n";
  exit;
}

#-- SQL command
my $qry = 'select city,population from canada_city where population>100000 order by population desc';

#-- Open the "mysql" client utility with the -B option to return
#-- results in batch tab-delimited format

my $command = "mysql -B -D ${database} -e '${qry}'";

my $line;
my @row;
my @fields;
my $first='y';

print "Query result\n";
print '-'x50,"\n";

open(COMM,"${command}|");
while ($line=<COMM>)
{
  chop $line;
  #-- If first line, fetch field names
  if ($first eq 'y')
  {
    $first = 'n';
    @fields = split(/\t/,$line);
    print "Fields: @fields","\n";
  }

  #-- Otherwise fetch field values
  else
  {
    @row = split(/\t/,$line);
    print "Row: @row","\n";
  }
}
close(COMM);

exit;

File: mysql/sample_mysql.py
#!/usr/bin/python
import pymysql #pip install PyMySQL
import os
print("User:", os.environ['USER'])

user = os.environ['USER']

# Name of database to be accessed
database = "alopes_db";

# Get connection specs from your .my.cnf file
my_cnf = "/home/%s/.my.cnf" % user

db = pymysql.connect(db=database,read_default_file=my_cnf)    

# prepare a cursor object using cursor() method
cur = db.cursor()

# Use all the SQL you like
cur.execute("select city, population from canada_city where population > 100000 order by population desc")

# print all the first cell of all the rows
for row in cur.fetchall():
    print("City: ", row[0], " - Population:", row[1])
db.close()
